src/mpid/ch4/netmod/ofi/ofi_win.h

```diff
@@ -962,8 +963,8 @@ static inline int MPIDI_NM_mpi_win_allocate_shared(MPI_Aint size,
     if (mpi_errno != MPI_SUCCESS)
         goto fn_fail;
 
-    sprintf(shm_key, "/mpi-%X-%X-%" PRIx64, MPIDI_Global.jobid, root_rank,
-            MPIDI_OFI_WIN(win).win_id);
+    snprintf(shm_key, MPIDI_OFI_MAX_SHMKEY_LEN, "/mpi-%X-%X-%" PRIx64, MPIDI_Global.jobid, root_rank,
```

Comment:
Actually, we may reduce the number of changes (e.g. we don't need
`MPIDI_MAX_SHMKEY_LEN` anymore, nor we need a new "errname" to
indicate that snprintf failed) if we have something like:

```
int shm_key_size;
...
shm_key = (char *) MPL_malloc(sizeof(char));
shm_key_size = snprintf(shm_key, 1, "/mpi-%s-%X-%" PRIx64, MPIDI_Global.jobid, root_rank,
                        MPIDI_OFI_WIN(win).win_id);
shm_key = (char *) MPL_realloc(shm_key, shm_key_size);
snprintf(shm_key, shm_key_size, "/mpi-%s-%X-%" PRIx64, MPIDI_Global.jobid, root_rank,
         MPIDI_OFI_WIN(win).win_id);
MPL_free(shm_key);
```

If we don't care about the following paragraph, We can avoid the first
malloc:

```
Concerning the return value of snprintf(), SUSv2 and C99 contradict
each other: when snprintf() is called with size=0 then SUSv2
stipulates an unspecified return value less than 1, while C99 allows
str to be NULL in this case, and gives the return value (as always) as
the number of characters that would have been written in case the
output string has been large enough.
```

If you still prefer to just fail, I can do that too :) I just thought
I'd suggest this before going ahead.